### DISCUSSION TITLE
Calendar how to set focus new date (new month)

### DISCUSSION THREAD

**Discussion by goodlythink** [mantinedev/mantine#1585] [Q&A] [answered]
![Screen Shot 2565-06-07 at 23 11 20](https://user-images.githubusercontent.com/12431734/172430214-8c5269d2-c557-4028-810f-b626154c01be.png)

if current date value 2022-06-07 and i set new date value to 2022-03-15
how to set focus to new date value

**Comment by rtivital** [comment, marked as answer]
1. Get ref of root calendar node: `<Calendar ref={calendarRef} />`
2. Select all buttons: `const buttons = calendarRef.current.querySelectorAll('button')`
3. Focus button with required value: `Array.from(buttons).find(button => button.textContent === dayToFocus.toString()).focus()`

**Reply by goodlythink** [reply]
thank you very much

**Reply by rhyek** [reply]
@rtivital how can we switch to a different month, directly? Your solution might work if I programmatically clicked the prev/next buttons until the desired month is visible. Might there be a different way?

**Reply by rtivital** [reply]
- Click on the level control in the header, it is between next and previous buttons
- Select month you want to switch to
- Click on that month.

**Reply by rhyek** [reply]
Thanks. I guess it would be great if the calendar used useImperativeHandle to export a method to control the focus/navigation. Because what if it's a different year, and so on.
